src/wasm.rs

```diff
@@ -103,25 +94,17 @@ fn handle_module(
     vprint!(flag_verbose, "Translating... ");
     let _ = terminal.reset();
 
-    let mut module_binary = read_to_end(path.clone()).map_err(|err| err.to_string())?;
-
-    if !module_binary.starts_with(&[b'\0', b'a', b's', b'm']) {
-        let mut features = Features::new();
-        if flag_enable_simd {
-            features.enable_simd();
-        }
-        if flag_enable_multi_value {
-            features.enable_multi_value();
-        }
-        if flag_enable_reference_types {
-            features.enable_reference_types();
-        }
-
-        module_binary = match wat2wasm_with_features(&module_binary, features) {
-            Ok(data) => data,
-            Err(e) => return Err(e.to_string()),
-        };
-    }
+    let module_binary = if path.to_str() == Some("-") {
+        let stdin = std::io::stdin();
+        let mut buf = Vec::new();
+        stdin
+            .lock()
+            .read_to_end(&mut buf)
+            .map_err(|e| e.to_string())?;
+        wat::parse_bytes(&buf).map_err(|err| format!("{:?}", err))?.into()
+    } else {
+        wat::parse_file(path).map_err(|err| format!("{:?}", err))?
```

Comment:
It's worth noting here that `wat` itself has the fallbacks already where if the input file is a wasm binary it won't attempt to parse it as text, it'll just return the raw binary that was read. When I removed the call to `read_to_end` here it made that a dead function, so I inlined the logic for handling `-`